`azurerm_storage_account_customer_managed_key`: update key vault uri document note for different subs (#27735)
hashicorp/terraform-provider-azurerm: website/docs/r/storage_account_customer_managed_key.html.markdown

* `key_vault_id` - (Optional) The ID of the Key Vault. Exactly one of `managed_hsm_key_id`, `key_vault_id`, or `key_vault_uri` must be specified.

> ~> Note: When the principal running Terraform has access to the subscription containing the Key Vault, it's recommended to use the `key_vault_id` property for maximum compatibility, rather than the `key_vault_uri` property. But if the Key Vault is in a different subscription from the Storage Account, use `key_vault_uri` instead; otherwise, using `key_vault_id` will cause drift.

Reviewer: ```suggestion
~> Note: When the principal running Terraform has access to the subscription containing the Key Vault, it's recommended to use the `key_vault_id` property for maximum compatibility, rather than the `key_vault_uri` property. However if the Key Vault is in a different subscription from the Storage Account `key_vault_uri` will need to be used instead otherwise there will be a diff.
```
